tests/DateTimeTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Psalm\Tests;
+
+use Psalm\Tests\Traits\ValidCodeAnalysisTestTrait;
+
+class DateTimeTest extends TestCase
+{
+    use ValidCodeAnalysisTestTrait;
+
+    /**
+     * @return iterable<string,array{string,assertions?:array<string,string>,error_levels?:string[]}>
+     */
+    public function providerValidCodeParse(): iterable
+    {
+        return [
+            'modify' => [
+                '<?php
+                    function getString(): string
+                    {
+                        return "";
+                    }
+
+                    $datetime = new DateTime();
+                    $dateTimeImmutable = new DateTimeImmutable();
+                    $a = $datetime->modify(getString());
+                    $b = $dateTimeImmutable->modify(getString());
+                    ',
+                'assertions' => [
+                    '$a' => 'DateTime|false',
+                    '$b' => 'DateTimeImmutable|false',
+                ],
+            ],
+            'modifyWithValidConstant' => [
+                '<?php
+                    /**
+                     * @return "+1 day"|"+2 day"
+                     */
+                    function getString(): string
+                    {
+                        return "+1 day";
+                    }
+
+                    $datetime = new DateTime();
+                    $dateTimeImmutable = new DateTimeImmutable();
+                    $a = $datetime->modify(getString());
+                    $b = $dateTimeImmutable->modify(getString());
+                    ',
+                'assertions' => [
+                    '$a' => 'DateTime',
+                    '$b' => 'DateTimeImmutable',
+                ],
+            ],
+            'modifyWithInvalidConstant' => [
+                '<?php
+                    /**
+                     * @return "kewk"|"koko"
```

Comment:
Can we stick to `"foo"`, `"bar"`, etc? I think using the widely known metasyntactic variables is a good idea in general so that it's more obvious that the value doesn't matter, or just use something like `"invalid modifier"|"also invalid"`. (Also, the first thing that turns up when you google "kewk" is urban dictionary...)

Reply:
changed